### PR TITLE
Locations determined from castle.json polygons on build

### PR DESCRIPTION
**Pull request by savage13**
Locations in the radar database are now determined from the castle.json polygons (GeoJSON format).

castle.json is the same between radar and objmap

Locations are determined using a point in 2d polygon (x-z) with a horizontal (flat) top and bottom in y.

Some polygons overlap (King's Study and Library) so a priority is used to identify which polygon takes precedence.

---
This change is [Reviewable](https://reviewable.io/reviews/zeldamods/radar/19)

### CODE REVIEW

**Comment by savage13** [comment]
Just a friendly reminder.  Not sure if this pull request got lost or not.

File: build.ts [hunk 1]
```diff
@@ -453,6 +456,100 @@ function korokGetType(group: any[], obj: any): string {
   process.exit(1);
 }
 
+// Check is a point (x,y,z) is contained within a polygon's bounding box
+//   Bounding box is defined in properties (xmin, xmax, zmin, zmax)
+function poly_point_in_bb(poly: any, pt: any): boolean {
```
Comment (leoetlino) [inline build.ts:461]: function names should be in camelCase (rather than snake_case)

File: build.ts [hunk 2]
```diff
@@ -453,6 +456,100 @@ function korokGetType(group: any[], obj: any): string {
...
+
+// Check is a point (x,y,z) is contained within a polygon
+//   The bounding box is first checked then the polygon is checked
+function poly_inside(poly: any, pt: any): boolean {
+  return poly_point_in_bb(poly, pt) && inside(pt, poly.geometry.coordinates[0]);
+}
+
+// Check if a point is within a polygon (pts)
+//  https://en.wikipedia.org/wiki/Point_in_polygon#Ray_casting_algorithm
+function inside(point: any, pts: any) {
```
Comment (leoetlino) [inline build.ts:477]: isPointInsidePolygon

File: build.ts [hunk 3]
```diff
@@ -453,6 +456,100 @@ function korokGetType(group: any[], obj: any): string {
...
+  let n = pts.length;
+  let xp = point[0];
+  let yp = point[2];
+  let xv: any = pts.map((p: any) => p[0]);
+  let yv: any = pts.map((p: any) => p[1]);
+
+  if (Math.abs(xv[0] - xv[n - 1]) < 1e-7 && Math.abs(yv[0] - yv[n - 1]) < 1e-7) {
+    n -= 1;
+  }
+  //console.log(xp, yp);
```
Comment (leoetlino) [inline build.ts:487]: this should be removed

File: build.ts [hunk 4]
```diff
@@ -453,6 +456,100 @@ function korokGetType(group: any[], obj: any): string {
...
+        nleft += 1;
+      }
+    }
+  }
+  let xin = nleft % 2;
+  return xin == 1;
+}
+
+// Test all polygons (polys) if a point lies in any
+//    polygons are assumed in GeoJSON format and have:
```
Comment (leoetlino) [inline build.ts:531]:
```suggestion
//    polygons are assumed to be in GeoJSON format and have:
```

File: build.ts [hunk 5]
```diff
@@ -453,6 +456,100 @@ function korokGetType(group: any[], obj: any): string {
...
+//      - a bounding box (xmin,ymin,zmin,xmax,ymax,zmax)
+//      - a priority where overlapping polygons with higher priority are choosen
```
Comment (leoetlino) [inline build.ts:533]:
```suggestion
//      - a priority where overlapping polygons with higher priority are chosen
```

File: build.ts [hunk 6]
```diff
@@ -453,6 +456,100 @@ function korokGetType(group: any[], obj: any): string {
...
+    const poly = polys.features[j];
+    if ((poly.properties.ymin && p[1] < poly.properties.ymin) ||
+      (poly.properties.ymax && p[1] > poly.properties.ymax)) {
+      continue;
+    }
+    if (found && poly.properties.priority < found.properties.priority) {
+      continue;
+    }
+    if (poly_inside(poly, p)) {
+      //console.log(`${p[0]} ${p[2]} ${j} ${obj.HashId} ${obj.UnitConfigName} ${poly.properties.name}`);
```
Comment (leoetlino) [inline build.ts:547]: this should be removed

**Review by savage13 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/zeldamods/radar/19)* status: 0 of 3 files reviewed, 6 unresolved discussions (waiting on @leoetlino)

___
*[`build.ts` line 461 at r1](https://reviewable.io/reviews/zeldamods/radar/19#-N1AsF0DJdkFR8xUO5LT-r1-461:-N2bQ19m6-K8RXymNd99:bil61ba) ([raw file](https://github.com/zeldamods/radar/blob/c16f68286531a3e58a8b1f97344f0a4ab6544645/build.ts#L461)):*
<details><summary><i>Previously, leoetlino (Léo Lam) wrote…</i></summary><blockquote>

function names should be in camelCase (rather than snake_case)
</blockquote></details>

Done. Changed to isPointInsideBoundingBox(). Also changed findPolygon(), isPointInsidePolygon(), and isPointInsidePolygonRCA() 
___
*[`build.ts` line 477 at r1](https://reviewable.io/reviews/zeldamods/radar/19#-N1AsF0DJdkFR8xUO5LT-r1-477:-N2bQC8AEreipnH9LdTn:b-896fix) ([raw file](https://github.com/zeldamods/radar/blob/c16f68286531a3e58a8b1f97344f0a4ab6544645/build.ts#L477)):*
<details><summary><i>Previously, leoetlino (Léo Lam) wrote…</i></summary><blockquote>

isPointInsidePolygon
</blockquote></details>

Done.
___
*[`build.ts` line 487 at r1](https://reviewable.io/reviews/zeldamods/radar/19#-N1AsF0DJdkFR8xUO5LT-r1-487:-N2bQGqS6b7RgY32CKYS:b-896fix) ([raw file](https://github.com/zeldamods/radar/blob/c16f68286531a3e58a8b1f97344f0a4ab6544645/build.ts#L487)):*
<details><summary><i>Previously, leoetlino (Léo Lam) wrote…</i></summary><blockquote>

this should be removed
</blockquote></details>

Done.
___
*[`build.ts` line 531 at r1](https://reviewable.io/reviews/zeldamods/radar/19#-N1AsF0DJdkFR8xUO5LT-r1-531:-N2bQLth6tM4MMeOwETv:b-896fix) ([raw file](https://github.com/zeldamods/radar/blob/c16f68286531a3e58a8b1f97344f0a4ab6544645/build.ts#L531)):*
<details><summary><i>Previously, leoetlino (Léo Lam) wrote…</i></summary><blockquote>

```suggestion
//    polygons are assumed to be in GeoJSON format and have:
```
</blockquote></details>

Done.
___
*[`build.ts` line 533 at r1](https://reviewable.io/reviews/zeldamods/radar/19#-N1AsF0DJdkFR8xUO5LT-r1-533:-N2bQQSO9zCZamNVNWz7:b-896fix) ([raw file](https://github.com/zeldamods/radar/blob/c16f68286531a3e58a8b1f97344f0a4ab6544645/build.ts#L533)):*
<details><summary><i>Previously, leoetlino (Léo Lam) wrote…</i></summary><blockquote>

```suggestion
//      - a priority where overlapping polygons with higher priority are chosen
```
</blockquote></details>

Done.
___
*[`build.ts` line 547 at r1](https://reviewable.io/reviews/zeldamods/radar/19#-N1AsF0DJdkFR8xUO5LT-r1-547:-N2bQW9G-HPOjulE6B2p:b-896fix) ([raw file](https://github.com/zeldamods/radar/blob/c16f68286531a3e58a8b1f97344f0a4ab6544645/build.ts#L547)):*
<details><summary><i>Previously, leoetlino (Léo Lam) wrote…</i></summary><blockquote>

this should be removed
</blockquote></details>

Done.

**Review by leoetlino — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/zeldamods/radar/19)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @savage13)

**Pull request merged**